scripts/main/designer/components/canvas/canvas.directive.js

```diff
@@ -12,9 +12,11 @@ angular.module('designer.module')
             link: function ($scope) {
                 $scope.selectedComponent = null;
 
-                $scope.selectFromCanvas = function (component) {
+                $scope.selectFromCanvas = function (component, $index) {
                     $scope.selectedComponent = component;
-                    $scope.onSelect({component: $scope.selectedComponent});
+                    $scope.onSelect({component: $scope.selectedComponent,
+                                     $index: $index
```

Comment:
looks like we don't need `$index`

Reply:
we are tracking all canvas elements by index and after clicking we can use appropriate model